client/v3/concurrency/session.go

```diff
@@ -47,7 +47,9 @@ func NewSession(client *v3.Client, opts ...SessionOption) (*Session, error) {
 
 	id := ops.leaseID
 	if id == v3.NoLease {
-		resp, err := client.Grant(ops.ctx, int64(ops.ttl))
+		ctx, cancel := context.WithTimeout(ops.ctx, time.Duration(ops.ttl)*time.Second)
```

Comment:
It is not appropriate to use `ttl` as timeout. Maybe we could add a default timeout constant. For example, `etcdctl` has a command timeout `defaultCommandTimeOut   = 5 * time.Second`